akka-stream-typed/src/main/scala/akka/stream/typed/scaladsl/ActorFlow.scala

```diff
@@ -151,4 +155,53 @@ object ActorFlow {
 
   }
 
+  /**
+   * Use the `ask` pattern to send a request-reply message to the target `ref` actor without including the context.
+   */
+  @implicitNotFound("Missing an implicit akka.util.Timeout for the ask() stage")
+  def askWithContext[I, Q, A, Ctx](ref: ActorRef[Q])(makeMessage: (I, ActorRef[A]) => Q)(
+      implicit timeout: Timeout,
+      ec: ExecutionContext): Flow[(I, Ctx), (A, Ctx), NotUsed] =
+    askWithContext(parallelism = 2)(ref)(makeMessage)(timeout, ec)
+
+  /**
+   * Use the `ask` pattern to send a request-reply message to the target `ref` actor without including the context.
+   */
+  @implicitNotFound("Missing an implicit akka.util.Timeout for the ask() stage")
+  def askWithContext[I, Q, A, Ctx](parallelism: Int)(ref: ActorRef[Q])(makeMessage: (I, ActorRef[A]) => Q)(
+      implicit timeout: Timeout,
+      ec: ExecutionContext): Flow[(I, Ctx), (A, Ctx), NotUsed] =
+    askImpl[(I, Ctx), Q, A, (A, Ctx)](parallelism)(ref)(
+      (in, r) => makeMessage(in._1, r),
+      (in, o: Future[A]) => o.map(a => a -> in._2))
```

Comment:
I think we can skip the implicit ec and use the same-thread `akka.dispatch.ExecutionContexts#parasitic` instead. That requires that it's never used for anything that throws, but I think it's only for mapping the result with the context from the tuple?

Reply:
Nice, I'm not used to having access to it!

Yes the EC is strictly used to reattach the context, so it should be safe to use